### PR TITLE
Read OpenStack creds from standard paths

### LINKED ISSUE #550: Verify openstackCredsFile
# Platform openstack:

With this PR, https://github.com/openshift/installer/pull/427 need to verify that the correct file to get OpenStack creds is this: `/etc/openstack/clouds.yaml`.  Another option would be `openStackCredsFile = os.Getenv("HOME") + "/.config/openstack"`

Once the OpenStack folks verify, this issue can be closed.    

With an OpenStack launch of installer, you will see a secret `openstack-creds` in `-n kube-system`  with encoded `clouds.yaml` in the data of the secret.  Might want to add multiple files/conditional to what creds file is found on the host system during launch in that secret.

### PR DESCRIPTION
**Pull request by flaper87**
OpenStack creds cold be in 3 different paths (etc, home config and
current dir). Instead of re-implementing the logic to find and read the
clouds.yaml file, we should use gophercloud which is the standard
go library for OpenStack.

Note that deployments on OpenStack are currently broken unless there's
a `clouds.yaml` under `/etc/openstack`.

Fixes #550

cc @tomassedovic @hardys @russellb @sallyom

### CODE REVIEW

**Comment by flaper87** [comment]
The failure in the `e2e-openstack` job is expected: Should be fixed by https://github.com/openshift/release/pull/2048 until a better solution is found.

/me working on it

**Comment by hardys** [comment]
+1 was discussing this issue today with @tomassedovic and we mentioned it'd be better to avoid paths completely in the installer and rely on gophercloud instead, thanks for working on it!

**Comment by flaper87** [comment]
FWIW, the `e2e-openstack` job failure is expected. The patch is doing its job as it's going past the previous failure "invalid argument"

**Comment by flaper87** [comment]
@sallyom @wking This should solve  #550 :tada:

**Comment by crawford** [comment]
Can you split this up into multiple commits? Vendoring operations should be in their own commit. Take a look at the repo's history for a sense of how we split things and write the commit messages.

**Comment by flaper87** [comment]
@crawford Thanks for the review! I've split the commits

**Comment by tomassedovic** [comment]
Strange. I am part of the openshift org. Guess I'm not a collaborator?

**Comment by tomassedovic** [comment]
@flaper87 thanks! I've tested it with and without the patch. Reproduced the issue and verified this patch fixes it.

So it's good to go from my end, but I don't have the rights to make it so, apparently.

**Comment by tomassedovic** [comment]
@crawford so FWIW, I've tested this change and I'm happy with merging it, but my powers don't reach that far. We'd appreciate you doing the honours if you've got no further objections.

**Comment by flaper87** [comment]
@tomassedovic already reviewed and tried to lgtm this patch. We've been added to the approvers list. I'll test the approve command on this PR to verify everything is working for us. (wonder if it'll let me approve my own patch)

/approve

**Comment by wking** [comment]
> @tomassedovic already reviewed and tried to lgtm this patch.

Maybe he needs to be in a team associated with this repo?  From [here][1]:

> changing LGTM is restricted to assignees, and only openshift/installer repo collaborators may be assigned issues.

GitHub will auto-complete the other `openstack-approvers` if I try to assign this issue manually, I think it's just something with @tomassedovic's account.

[1]: https://github.com/openshift/installer/pull/588#issuecomment-436612458

File: pkg/asset/manifests/tectonic.go [hunk 1]
```diff
@@ -80,10 +80,17 @@ func (t *Tectonic) Generate(dependencies asset.Parents) error {
 			},
 		}
 	case "openstack":
-		credsEncoded, err := credsFileEncode(openStackCredsFile)
```
Comment (wking) [inline pkg/asset/manifests/tectonic.go]: Looks like you're removing the last consumer here.  Can you drop `credsFileEncode` too?

Comment (flaper87) [reply pkg/asset/manifests/tectonic.go]: done, thanks a bunch!

**Comment by flaper87** [comment]
> > @tomassedovic already reviewed and tried to lgtm this patch.
> 
> Maybe he needs to be in a team associated with this repo? From [here](https://github.com/openshift/installer/pull/588#issuecomment-436612458):

We're all part of this team: https://github.com/orgs/openshift/teams/team-openstack-cloud-provider

I'll verify whether that team is associated with this repo. If so, then it's something related to @tomassedovic account as you said :D

**Comment by flaper87** [comment]
According to https://github.com/orgs/openshift/teams/team-openstack-cloud-provider/repositories?utf8=%E2%9C%93&query= the team is not associated with this repo. @smarterclayton can you please add this repo to our team? While you're at it, it'd be great to also add release, MCO, and MAO and other repos you think we might need to provide reviews on :)

**Comment by flaper87** [comment]
Let's try again

/approve
/lgtm

**Comment by flaper87** [comment]
> @flaper87: you cannot LGTM your own PR.
> 
> In response to [this](https://github.com/openshift/installer/pull/588#issuecomment-436963651):
> 
> > Let's try again
> > /approve
> > /lgtm
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.

Smart bot, makes sense! :smile:

**Comment by wking** [comment]
Hooray for both the merge and @tomassedovic becoming assignable ;).  For future reference, @tomassedovic do you know what changed for your to become assignable?

**Comment by tomassedovic** [comment]
@wking

@eparis fixed it for me. He said: "/assign requires you to have read access to the repo, not just the repo be public...."

**Pull request merged**